docs/.Doxyfile

```diff
@@ -341,6 +369,17 @@ MARKDOWN_SUPPORT       = YES
 
 TOC_INCLUDE_HEADINGS   = 0
 
+# The MARKDOWN_ID_STYLE tag can be used to specify the algorithm used to
+# generate identifiers for the Markdown headings. Note: Every identifier is
+# unique.
+# Possible values are: DOXYGEN use a fixed 'autotoc_md' string followed by a
+# sequence number starting at 0 and GITHUB use the lower case version of title
+# with any whitespace replaced by '-' and punctuation characters removed.
+# The default value is: DOXYGEN.
+# This tag requires that the tag MARKDOWN_SUPPORT is set to YES.
+
+MARKDOWN_ID_STYLE      = DOXYGEN
```

Comment:
This comment seems to suggest a new feature controlling how anchors for headers are generated for Markdown files. Would this have solved our problem?

Reply:
I think it might. I'll merge this so CI can pass for other PRs with red CI and create a new PR to address configuring Doxygen more appropriately.